test/e2e/validation_test.go

```diff
@@ -0,0 +1,188 @@
+func getInstanceID(providerID string) string {
+	providerTokens := strings.Split(providerID, "/")
+	return providerTokens[len(providerTokens)-1]
+}
+
+// getInstanceIDsCreated returns the instanceIDs of all the Windows nodes created
+func getInstanceIDsCreated(kubeclient kubernetes.Interface) ([]string, error) {
+	nodes, err := kubeclient.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: wmc.WindowsOSLabel})
+	if err != nil {
+		return nil, errors.Wrap(err, "error while querying for Windows nodes")
+	}
+	instanceIDs := make([]string, 0, len(nodes.Items))
+	for _, node := range nodes.Items {
+		if len(node.Spec.ProviderID) > 0 {
+			instanceID := getInstanceID(node.Spec.ProviderID)
+			instanceIDs = append(instanceIDs, instanceID)
+		}
+	}
+	return instanceIDs, nil
+}
+
+// testConfigMapValidation ensures that the required configmap is created and is having appropriate
+// entries
+func testConfigMapValidation(t *testing.T, nodeCount int) {
+	testCtx := framework.NewTestCtx(t)
+	namespace, err := testCtx.GetNamespace()
+	require.NoError(t, err, "error while getting test namespace")
+
+	err = waitForTrackerConfigMap(framework.Global.KubeClient, namespace, nodeCount,
+		retryInterval, time.Minute*1)
+	require.NoError(t, err, "error waiting for tracker configmap")
+
+	// Get the instance id from the cloud provider for the windows Nodes created
+	instanceIDs, err := getInstanceIDsCreated(framework.Global.KubeClient)
+	require.NoError(t, err, "error while getting provider specific instanceIDs")
+
+	// check if those instances are present in the configmap
+	trackerConfigMap, err := framework.Global.KubeClient.CoreV1().ConfigMaps(namespace).Get(tracker.StoreName, metav1.GetOptions{})
+	require.NoError(t, err, "error while getting the tracker configmap")
+	for _, instanceID := range instanceIDs {
+		require.Contains(t, trackerConfigMap.BinaryData, instanceID)
+	}
+}
+
+// getExpectedNodeCreds instantiates the cloud provider and gets the credentials to access it. Since we're not
+// exposing the WindowsVMs created in any of the CRs, we have to validate the credentials via values we get from
+// cloud provider.
+// TODO: This will be removed once we have construct node object and have some functionality to properly test.
+func validateSSHConnection(ipAddress string, credentials tracker.Credentials) error {
+	config := &ssh.ClientConfig{
+		User:            credentials.Username,
+		Auth:            []ssh.AuthMethod{ssh.Password(credentials.Password)},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	sshClient, err := ssh.Dial("tcp", ipAddress+":22", config)
+	if err != nil {
+		return errors.Wrap(err, "failed to establish SSH connection")
+	}
+	session, err := sshClient.NewSession()
+	if err != nil {
+		return errors.Wrap(err, "failed to establish SSH session")
+	}
+	err = session.Run("dir")
+	if err != nil {
+		return errors.Wrap(err, "failed to run dir command on remote Windows VM")
+	}
```

Comment:
Why are you doing this? Isn't better to get WindowsVM interface and use that instead? Doing this has the danger of everyone else following this pattern.